lib/membrane/video_compositor/compound.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Membrane.VideoCompositor.Compound do
+  @moduledoc """
+  Structure representing Compound objects.
+
+  Compound objects can be used to combine multiple inputs
+  into one object, by applying `Membrane.VideoCompositor.Compound.Layout`
+  to them, according to the `inputs_map` field.
+  """
+
+  alias Membrane.VideoCompositor.Object
+  alias __MODULE__.{BaseSize, Layout}
+
+  @enforce_keys [:base_size, :inputs_map, :layout]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          # base_size: BaseSize.t() | Object.input_t(),
+          base_size: BaseSize.t(),
+          inputs_map: %{Object.name_t() => Layout.input_t()},
+          layout: Layout.name_t()
+        }
+end
```

Comment:
What `Compounds` can take as input? Maybe we should have some way of specifying whether inputs should be `Texture`, `Canvas`, or other `Compounds` types? Otherwise, I can't think of a simple way of specifying what type of input is accepted by certain `Compound`.

Reply:
The idea was to always render objects as a texture after all transformations adn pass it to the next objects. Compounds de facto will always take a texture as an input.